components/style/sharing/mod.rs

```diff
@@ -109,6 +218,103 @@ pub struct StyleSharingCandidateCache<E: TElement> {
     cache: LRUCache<StyleSharingCandidate<E>>,
 }
 
+/// A struct we use as a simple cache of stuff we use to test against different
+/// candidates.
+#[derive(Debug)]
+pub struct ElementSharingChecker<E: TElement> {
```

Comment:
I think it's nice because it allows you to keep the sharing flags stuff in a well-known comment, but I need to rewrite all this anyway, so we'll see :)